subprojects/core/src/main/java/org/gradle/internal/exceptions/encryption/EncryptionException.java

```diff
@@ -0,0 +1,33 @@
+/*
+ * Copyright 2024 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.internal.exceptions.encryption;
+
+import org.gradle.api.GradleException;
+import org.gradle.internal.exceptions.Contextual;
+
+import javax.annotation.Nullable;
+
+/**
+ * This exception can be thrown when an error occurs during the setup or execution of encryption work.
+ */
+@Contextual
+public class EncryptionException extends GradleException {
```

Comment:
ℹ️  The exception class cannot live in `encryption-services` module, because it has to be on the runtime classpath of the Gradle client, as the daemon forwards the exception as is.

This is the error that happens if the exception class is located in `encryption-services`:
```
Caused by: java.lang.ClassNotFoundException: org.gradle.internal.encryption.impl.EncryptionException
```

<details>
<summary>Full stacktrace</summary>
```
org.gradle.launcher.daemon.client.DaemonConnectionException: Could not receive a message from the daemon.	
	at org.gradle.launcher.daemon.client.DaemonClientConnection.receive(DaemonClientConnection.java:88)	
	at org.gradle.launcher.daemon.client.DaemonClient.monitorBuild(DaemonClient.java:245)	
	at org.gradle.launcher.daemon.client.DaemonClient.executeBuild(DaemonClient.java:208)	
	at org.gradle.launcher.daemon.client.DaemonClient.execute(DaemonClient.java:158)	
	at org.gradle.launcher.daemon.client.DaemonClient.execute(DaemonClient.java:97)	
	at org.gradle.launcher.cli.RunBuildAction.run(RunBuildAction.java:56)	
	at org.gradle.internal.Actions$RunnableActionAdapter.execute(Actions.java:167)	
	at org.gradle.launcher.cli.DefaultCommandLineActionFactory$ParseAndBuildAction.execute(DefaultCommandLineActionFactory.java:332)	
	at org.gradle.launcher.cli.DefaultCommandLineActionFactory$ParseAndBuildAction.execute(DefaultCommandLineActionFactory.java:296)	
	at org.gradle.launcher.cli.DebugLoggerWarningAction.execute(DebugLoggerWarningAction.java:74)	
	at org.gradle.launcher.cli.DebugLoggerWarningAction.execute(DebugLoggerWarningAction.java:30)	
	at org.gradle.launcher.cli.WelcomeMessageAction.execute(WelcomeMessageAction.java:97)	
	at org.gradle.launcher.cli.WelcomeMessageAction.execute(WelcomeMessageAction.java:40)	
	at org.gradle.launcher.cli.NativeServicesInitializingAction.execute(NativeServicesInitializingAction.java:50)	
	at org.gradle.launcher.cli.NativeServicesInitializingAction.execute(NativeServicesInitializingAction.java:27)	
	at org.gradle.launcher.cli.ExceptionReportingAction.execute(ExceptionReportingAction.java:41)	
	at org.gradle.launcher.cli.ExceptionReportingAction.execute(ExceptionReportingAction.java:26)	
	at org.gradle.launcher.cli.DefaultCommandLineActionFactory$WithLogging.execute(DefaultCommandLineActionFactory.java:444)	
	at org.gradle.launcher.Main.doAction(Main.java:35)	
	at org.gradle.launcher.bootstrap.EntryPoint.run(EntryPoint.java:52)	
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)	
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)	
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)	
	at java.lang.reflect.Method.invoke(Method.java:498)	
	at org.gradle.launcher.bootstrap.ProcessBootstrap.runNoExit(ProcessBootstrap.java:72)	
	at org.gradle.launcher.bootstrap.ProcessBootstrap.run(ProcessBootstrap.java:39)	
	at org.gradle.launcher.GradleMain.main(GradleMain.java:31)	
Caused by: org.gradle.internal.remote.internal.RecoverableMessageIOException: Could not read message from '/127.0.0.1:34963'.	
	at org.gradle.internal.remote.internal.inet.SocketConnection.receive(SocketConnection.java:95)	
	at org.gradle.launcher.daemon.client.DaemonClientConnection.receive(DaemonClientConnection.java:82)	
	... 26 more	
Caused by: java.lang.ClassNotFoundException: org.gradle.internal.encryption.impl.EncryptionException	
	at java.net.URLClassLoader.findClass(URLClassLoader.java:387)	
	at java.lang.ClassLoader.loadClass(ClassLoader.java:418)	
	at java.lang.ClassLoader.loadClass(ClassLoader.java:351)	
	at java.lang.Class.forName0(Native Method)	
	at java.lang.Class.forName(Class.java:348)	
	at java.io.ObjectInputStream.resolveClass(ObjectInputStream.java:763)	
	at org.gradle.internal.serialize.ClassLoaderObjectInputStream.resolveClass(ClassLoaderObjectInputStream.java:42)	
	at java.io.ObjectInputStream.readNonProxyDesc(ObjectInputStream.java:2029)	
	at java.io.ObjectInputStream.readClassDesc(ObjectInputStream.java:1893)	
	at java.io.ObjectInputStream.readOrdinaryObject(ObjectInputStream.java:2227)	
	at java.io.ObjectInputStream.readObject0(ObjectInputStream.java:1710)	
	at java.io.ObjectInputStream.defaultReadFields(ObjectInputStream.java:2472)	
	at java.io.ObjectInputStream.defaultReadObject(ObjectInputStream.java:638)	
	at java.lang.Throwable.readObject(Throwable.java:915)	
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)	
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)	
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)	
	at java.lang.reflect.Method.invoke(Method.java:498)	
	at java.io.ObjectStreamClass.invokeReadObject(ObjectStreamClass.java:1185)	
	at java.io.ObjectInputStream.readSerialData(ObjectInputStream.java:2363)	
	at java.io.ObjectInputStream.readOrdinaryObject(ObjectInputStream.java:2254)	
	at java.io.ObjectInputStream.readObject0(ObjectInputStream.java:1710)	
	at java.io.ObjectInputStream.defaultReadFields(ObjectInputStream.java:2472)	
	at java.io.ObjectInputStream.defaultReadObject(ObjectInputStream.java:638)
```
</details>